frontend/src/pages/ProgressionChecker/CoursesModal/CoursesModal.tsx

```diff
@@ -17,6 +17,12 @@ const CoursesModal = ({
   title, courses, modalVisible, setModalVisible,
 }: Props) => {
   const [sortFn, setSortFn] = useState(SortFn.AlphaNumeric);
+  const [filter, setFilter] = useState('');
+
+  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value: inputValue } = e.target;
+    setFilter(inputValue);
```

Comment:
can just do `setFilter(e.target.value)` rather than destructing the variable since the var is only used once. saves one line ;)

Reply:
oop im dumb